### PR TITLE
combobox/dropdown - changed to match new rounded designs

### PR DESCRIPTION
**Pull request by agliga**
## Description
Changed combobox to match the new designs 
Changed dropdown as well

## Context
* Combobox now supports floating label
* Changed combobox colors to match designs
* Made dropdown to have a `4px` space between the element that's opening it
* Rounded the first and the last item's highlight in dropdown
* Fixed fake-menu to be rounded (This was a special case and required some refactoring. See below)

### Fake menu changes
This was an interesting one. I had added a change to detect `first-child` and `last-child` to check to see to round the top/bottom part only for each highlighted item
However for fake-menu the syntax is like this:
```
<ul .menu>
  <li> <a .option>Anchor</a></li>
  <li> <a .option>Anchor</a></li>
</ul>
```
So doing `first-child` on option was not working since it needed to be on the `li`. I basically split up the rounding mixins part to apply the mixins on the `li`, but the other highlights and such on the anchor. Seems to work well

## References
#1256 

## Screenshots
Screen Shot 2020-11-12 at 1 53 23 PM
Screen Shot 2020-11-12 at 1 53 26 PM
Screen Shot 2020-11-12 at 1 53 34 PM
Screen Shot 2020-11-12 at 1 53 39 PM
Screen Shot 2020-11-12 at 1 54 06 PM
Screen Shot 2020-11-12 at 1 54 25 PM
Screen Shot 2020-11-12 at 1 54 43 PM

### CODE REVIEW

**Comment by agliga** [comment]
> I think the focussed state background colour in dark mode is supposed to be black. Everything else is looking great though.
> 
> 
> 
> Screen Shot 2020-11-13 at 10 54 26 AM
> 
> 

Yes i was planning on having another pr for all dark mode. The colors are not right for it. I think textbox would need it as well
I did not tackle it in this pr but was planning on doing it in another one

**Comment by ianmcburnie** [comment]
Sounds good, Andrew. Will approve this one...

**Review by ianmcburnie — APPROVED** [review]

File: dist/combobox/ds6/combobox.css [hunk 1]
```diff
@@ -95,6 +95,18 @@ span.combobox {
 .combobox__option[role^="option"]:active {
   font-weight: bold;
 }
+.combobox__option[role^="option"]:first-child {
```
Comment (austingardner) [inline dist/combobox/ds6/combobox.css:98]: what is the `[role^="option"]`  syntax called again? I need to learn more about this

**Review by austingardner — APPROVED** [review]
LGTM

**Pull request merged**